### DISCUSSION TITLE
How to submit a new workflow from WorkflowTemplate?

### DISCUSSION THREAD

**Discussion by mingjliu9** [argoproj/argo-workflows#8326] [Q&A] [answered]
Is there an example to submit a new workflow with parameters(e.g., `{"paramA":"valueA"}`) from created workflowTemplates?
I tried to read https://raw.githubusercontent.com/argoproj/argo-workflows/master/api/openapi-spec/swagger.json but I didn't see how to referrer to existing workflow template.
Thank you!

**Comment by terrytangyuan** [comment]
You can use `--from` to submit a single workflow from an existing resource:

```sh
argo submit --from cronwf/my-cron-wf
```

**Reply by alexec** [reply, marked as answer]
```sh
argo submit —from wftmpl/my-template -p my-param=my-value
```

**Reply by szEvEz** [reply]
Is this also possible via the REST API?

**Reply by wesleyscholl** [reply]
Yes it's possible - Send a POST request to [https://localhost:2746/api/v1/workflows/{namespace}/submit](https://localhost:2746/api/v1/workflows/%7Bnamespace%7D/submit)

Sending this JSON in the body:
```json
{
  "resourceKind": "WorkflowTemplate",
  "resourceName": "workflow-template-name",
  "submitOptions": {
    "parameters": ["param1=1234", "param2=abcd"]
  }
}
```
Reference: https://github.com/argoproj/argo-workflows/issues/3712

**Reply by RothAndrew** [reply]
What is the difference between `POST /api/v1/workflows/{namespace}/submit` and `POST /api/v1/workflows/{namespace}`? One is named `SubmitWorkflow` in the spec and the other is named `CreateWorkflow`.
